### PR TITLE
Fix unit tests in 3.0 (Work In Progress)

### PR DESCRIPTION
**Pull request by jorgeatorres**
3.0 version of the changes in PR #75.

### CODE REVIEW

File: tests/base/FrmUnitTest.php [hunk 1]
```diff
@@ -28,6 +28,21 @@ function setUp() {
 			delete_option('frm_db_version');
 		}
 
+		// Some of the tests for FrmDb are triggering a transaction commit, preventing further tests from working.
+		// This is a temporary workaround until we review FrmDb tests in detail.
+		global $wpdb;
+		if ( $wpdb->get_var( "SHOW TABLES LIKE '{$wpdb->prefix}frm_fields'" ) )
```
Comment (stephywells) [inline tests/base/FrmUnitTest.php]: Can you add braces to help us follow the WP code styling?
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#brace-style

File: tests/emails/test_FrmProEmailWithAttachment.php [hunk 2]
```diff
@@ -101,7 +101,10 @@ private function check_for_multi_part_message( $body ) {
 	private function check_single_file_upload_attachment( $body ) {
 		$file_field_id = FrmField::get_id_by_key( 'file_upload_single' );
 		$media_id = $this->entry->metas[ $file_field_id ];
-		$file_name = FrmProFieldsHelper::get_displayed_file_html( $media_id, 'thumbnail', array( 'show_filename' => true ) );
+
+		$file_field = FrmFieldFactory::get_field_type( 'file' );
+		$file_name = $file_field->get_displayed_file_html( $media_id, 'thumbnail', array( 'show_filename' => true ) );
+		// $file_name = FrmProFieldsHelper::get_displayed_file_html( $media_id, 'thumbnail', array( 'show_filename' => true ) );
```
Comment (stephywells) [inline tests/emails/test_FrmProEmailWithAttachment.php]: Is there a reason this line is sticking around?

File: classes/helpers/FrmAppHelper.php [hunk 3]
```diff
@@ -180,7 +180,13 @@ public static function prevent_caching() {
...
      */
     public static function is_admin() {
-        return is_admin() && ( ! defined('DOING_AJAX') || ! DOING_AJAX );
+		if ( function_exists( 'wp_doing_ajax' ) ) {
+			$doing_ajax = wp_doing_ajax();
+		} else {
+			$doing_ajax = defined('DOING_AJAX') && DOING_AJAX;
+		}
+
+	    return is_admin() && ! $doing_ajax;
```
[the remark this answers is not in the document]
Comment (jorgeatorres) [reply classes/helpers/FrmAppHelper.php]: Much better. I wanted to add the code to `doing_ajax()` but wasn't sure due to the `self::is_preview_page()` call. The new function is certainly better.

File: tests/base/FrmUnitTest.php (hunk 1 above)
Comment (jorgeatorres) [reply tests/base/FrmUnitTest.php]: Fixed.

File: tests/emails/test_FrmProEmailWithAttachment.php (hunk 2 above)
Comment (jorgeatorres) [reply tests/emails/test_FrmProEmailWithAttachment.php]: No reason. Just wanted to confirm it was working before removing it.

**Pull request merged**